### DISCUSSION TITLE
New install questions

### DISCUSSION THREAD

**Discussion by mallison01** [xblax/flashforge_ad5m_klipper_mod#74] [General]
Just installed today, but have a few questions.

Installed the version with klipperscreen on a AD5M with the flashforge 5m series camera.

I have done one quick test print with orca and it seems good.

Is there a way to change the time zone, mainly whats displayed on the screen?

Any chance of installing moonraker timelapse?

Or Obico?

Thanks, 
Marty

**Comment by consp** [comment]
> Is there a way to change the time zone, mainly whats displayed on the screen?

Currently that is not possible. I'll have a look if it's possible to add a TZ on the display. Shouldn't be too hard. It will be a setting in the klipperscreen settings or something equivalent.

> Any chance of installing moonraker timelapse?

Highly unlikely. Moonraker timelapse uses ffmpeg which i've had crash the entire system. It needs too much memory. A rpi, which it uses as an example, has an order of magnitude more memory unfortunately. I'll have a look and see what it exactly does but don't get your hopes up.

> obico

From the docs:

> If it has at least 4gb of RAM, it should run fine.

We have 128 _MB_ (and also lack a gpu).

**Reply by mallison01** [reply]
As far as moonraker timelapse, i kinda figured that may be the case. Is there anyway to offload this to another machine (vm, docker etc)

As for obico, i think you looked at the self hosted server requirements, it does want 4gb of ram and prefers a gpu. I have this running in docker on a very beefy server.

i was interested in running the client, similar to octoeverywhere..

https://www.obico.io/docs/user-guides/moonraker-obico/overview/

I do not see any requirements listed for the client.

**Reply by consp** [reply]
That might be possible. Doesn't look too special even if a bit bloaty. The scripts won't work though but the main program is just python. One worrying thing is that it does process the webcam stream, while it does not re-encode it it does chunk it (eating memory).

I'd say this is something for when we get the creation of virtual-env's streamlined, currently adding a package is quite a bit of work.  Maybe it's one of those things which can become a separate package or something so people can decide on their own.

----
Looking at the requirements there is this though:
```
sentry-sdk==1.23.1
```
This is just me personally, but I do not think you want a sentry client running on such a small device. Besides the fact it phones home, even worse the default is opt _IN_ and the default url is a company url.  Sentry sends a lot if not everything it thinks it needs and is (in most common settings) a GDPR violation waiting to happen.

**Reply by mallison01** [reply]
Would it be possible to offload any of these kind of things to a container or piggyback a raspberry pi or similar to the printer?

As for the phone home, I wonder if this set to the default cloud service or is changed when you switch to self hosted?

If this is something I can help with or if there is anything I can help with please let me know.

If I could get this tied into my self hosted obico server that would make this perfect for me.

I've already ran a few prints and they were great, all I had to do was change the start and end gcode as mentioned and connect orca slicer to the printer and it's been good so far.

Klipperscreen is working great so far.

I've added the flashforge camera they sell and added my own led lights plugged into the led port on the mb and they are now controlled.

I have printed and installed the diy enclosure, I just need the sides for it.

I may also try to add the chamber fans. 

Thanks for all the work so far.

**Reply by consp** [reply]
> If I could get this tied into my self hosted obico server that would make this perfect for me.

Should be testable, also solves the sending home problem :)

I think the main goal now is getting it stable and making it easier for people to use (and help, see also #32). Once that is done I guess it would be easier to add components and it's possible to consider adding stuff. There is no real "make your own python part" guide yet but I'm pretty sure someone will bite the pill and make that after a while.

> Klipperscreen is working great so far.

Good to hear, since that is mainly my controbution 😊

> I may also try to add the chamber fans.

Is that possible with the non-pro? I thought the mosfets for that were missing or did you buy a pro controller board?

**Reply by mallison01** [reply]
It looks great so far, I'll print some more with it this weekend and week.

Klipperscreen is great, only thing I had was trying to figure out a way to change timezones, which I see was added as an issue.

If there is anything I can do or test let me know.

As far as the chamber fan. I do not have a pro board. I have seen something about some missing components. Somewhere I did see someone print a duct system for a fan but I didn't look into too much at the time.

If all we are missing are some minor components I wouldn't mind soldering them on to add the features now that it's open to control it.

**Reply by mallison01** [reply]
I did notice the Unload filament macro was missing, i added it my self and works fine. I do see an issue listed for this.

**Reply by consp** [reply]
> I do see an issue listed for this.

It's an open issue to add it or not, you could post it there.
